corehq/apps/data_interfaces/pillow.py

```diff
@@ -59,23 +58,24 @@ def _is_system_change(change):
 
         return False
 
-    def _get_applicable_rules(self, change):
+    def _get_applicable_rules(self, change, case):
         domain = change.metadata.domain
-        associated_form_id = change.metadata.associated_document_id
+        form_id = change.metadata.associated_document_id
 
         # TODO: feels like there should be some enforced order for running through rules?
         rules = self._get_rules(domain)
 
-        if not associated_form_id or associated_form_id == UPDATE_REASON_RESAVE:
+        if not form_id or form_id == UPDATE_REASON_RESAVE:
             # no associated form occurs whenever a form is rebuilt. Forms can be rebuilt
             # when a deletion is being undone or a form is being restored. In either case,
             # all rules may be interested in these changes
             applicable_rules = rules
         else:
             associated_form = self._get_associated_form(change)
             if associated_form and not is_dedupe_xmlns(associated_form.xmlns):
-                case_updates = get_case_updates(associated_form, for_case=change.id)
-                applicable_rules = [rule for rule in rules if self._has_applicable_changes(case_updates, rule)]
+                case_properties = set(case.case_json) | {"external_id", "owner_id", "name", "date_opened"}
```

Comment:
Should this list of system properties that are available to use in a rule definition live in a shared place to ensure consistency between the front and backend?

Reply:
2d81047d08624c8eb6b86d39c6bc875723943eca